lib/board.rb

```diff
@@ -0,0 +1,74 @@
+class Board
+
+  WIN_COMBINATIONS = [[0, 1, 2], 
+                      [3, 4, 5],
+                      [6, 7, 8],
+                      [0, 3, 6],
+                      [1, 4, 7],
+                      [2, 5, 8],
+                      [0, 4, 8],
+                      [2, 4, 6]]
+
+  attr_accessor :cells
+
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(user_input)
+    cells[user_input.to_i - 1]
+  end
+
+  def full?
+    !cells.include?(" ")
+  end
+
+  def turn_count
+    9 - cells.count(" ")
+  end
+
+  def taken?(position)
+    cells[position.to_i - 1] != " "
+  end
+
+  def valid_move?(user_input)
+    if !(1..9).include?(user_input.to_i)
+      return false
+    end
+
+    !taken?(user_input)
+  end
```

Comment:
Use logical and operator (`&&`) to combine multiple logical expressions and utilize their implicit return value.
```
(1..9).include?(user_input.to_i) && !taken(user_input)
```